RecoTracker/LSTCore/src/alpaka/Event.dev.cc

```diff
@@ -286,7 +285,6 @@ void lst::Event<Acc3D>::addPixelSegmentToEvent(std::vector<unsigned int> const&
     mdsInGPU->setData(*miniDoubletsBuffers);
 
     alpaka::memcpy(queue, miniDoubletsBuffers->nMemoryLocations_buf, nTotalMDs_view);
-    alpaka::wait(queue);
```

Comment:
@makortel
I removed the `wait` here, but I have second thoughts: isn't this problematic because `    unsigned int nTotalMDs;` as well as `auto nTotalMDs_view` can disappear after we go out of scope on the next line and `alpaka::memcpy` can fail since it may run later.

Reply:
> isn't this problematic because ` unsigned int nTotalMDs;` as well as `auto nTotalMDs_view` can disappear after we go out of scope on the next line and `alpaka::memcpy` can fail since it may run later.

I think you are correct. You could avoid the `wait()` with pinned host memory buffer from the caching allocator
```cpp
auto nTotalMDs_buf = cms::alpakatools::make_host_buffer<unsigned int>(queue);
// replace uses of nTotalMDs and nTotalMDs_view with nTotalMDs_buf

alpaka::memcpy(queue, miniDoubletsBuffers->nMemoryLocations_buf, nTotalMDs_buf);
```
(this would be safe even in case the caching would be disabled, because then the `nTotalMDs_buf` destructor would synchronize)

Is the device-to-host copy needed for the memory allocation via `lst::MiniDoubletsBuffer` constructor on line 284?